### PR TITLE
Update epw-viz

### PR DESCRIPTION
**Pull request by devngc**
Each chart is cached separately in this revision.

### CODE REVIEW

File: epw-viz/app.py [hunk 1]
```diff
@@ -2,642 +2,419 @@
...
+        # A dictionary of EPW variable name to its corresponding field number
+        fields = get_fields()
+
+        # epw file #####################################################################
+        with st.expander('Upload EPW file'):
+            epw_data = st.file_uploader('', type='epw')
+            if epw_data:
+                epw_file = pathlib.Path('./data/sample.epw')
+                epw_file.parent.mkdir(parents=True, exist_ok=True)
+                epw_file.write_bytes(epw_data.read())
```
Comment (mostaphaRoudsari) [inline epw-viz/app.py:41]: Should we not try to save the file based on the filename to ensure one user doesn't overwrite another user's file if they use the app at the same time?

**Review by mostaphaRoudsari — APPROVED** [review]
Hi @devngc! Thank you for the update. I'm going to merge this in but I added a comment that I think we need to address at some point to make the app really useful for several users at a time.

File: epw-viz/app.py (hunk 1 above)
Comment (devngc) [reply epw-viz/app.py:41]: PR [here](https://github.com/pollination/sample-apps/pull/31)

**Comment by mostaphaRoudsari** [comment]
Thanks @devngc! I'm going to revert all the changes here. The caching is not implemented correctly and breaks the app.

Here is what I get for New York after loading a weather file from Australia first.

image

I should have tested it first on my side before merging. I thought you have already done the basic checks. As I said on Basecamp, let's forget about refactoring this app and move to build the real one.

**Comment by devngc** [comment]
This is something I am aware of. We have to remove caching from the image crawler to make this work. Or else, we could simply remove the image crawler.

**Comment by devngc** [comment]
What's the error here?

**Comment by mostaphaRoudsari** [comment]
The charts are not being updated - the issue is not with the image.

**Comment by devngc** [comment]
Can you show me on a call? It work fine at my end.

**Pull request merged**